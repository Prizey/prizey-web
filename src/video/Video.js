import React, { useState } from 'react'
import ReactPlayer from 'react-player'
import { navigate } from '@reach/router'
import styles from './styles'

export const navigateTo = videoRedirectUrl => () => navigate(videoRedirectUrl)

export const updateProgressBar = setPlayed => ({ played }) =>
  setPlayed(played * 100)

const Video = ({ videoEmbedUrl, videoRedirectUrl, videoText }) => {
  const [played, setPlayed] = useState(0)

  return (
    <div style={styles.app}>
      <div style={styles.wrapper}>
        <ReactPlayer
          style={styles.player}
          url={videoEmbedUrl}
          playing
          onEnded={navigateTo(videoRedirectUrl)}
          onProgress={updateProgressBar(setPlayed)}
          width="100%"
          height="100%"
        />
        <div style={styles.text}>{videoText}</div>
        <div style={styles.progressBar}>
          <div style={{ ...styles.progress, width: `${played}%` }} />
        </div>
      </div>
    </div>
  )
}

export default Video
