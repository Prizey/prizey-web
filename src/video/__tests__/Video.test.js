import React from 'react'
import renderer from 'react-test-renderer'
import { navigate } from '@reach/router'

import Video, { navigateTo, updateProgressBar } from '../Video'

jest.mock('react-player', () => props => (
  <div {...props}>react-player - {props.children}</div>
))

jest.mock('@reach/router', () => ({
  navigate: jest.fn(),
}))

it('renders correctly', () => {
  const props = {
    videoEmbedUrl: 'Foo videoEmbedUrl',
    videoRedirectUrl: 'Foo videoRedirectUrl',
    videoText: 'Foo videoText',
  }
  const tree = renderer.create(<Video {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe('navigateTo', () => {
  it('works correctly', () => {
    const videoRedirectUrl = 'https://foo-video-redirect-url.cc'

    navigateTo(videoRedirectUrl)()

    expect(navigate).toHaveBeenCalledWith(videoRedirectUrl)
  })
})

describe('updateProgressBar', () => {
  it('works correctly', () => {
    const setPlayed = jest.fn()

    updateProgressBar(setPlayed)({ played: 1 })

    expect(setPlayed).toHaveBeenCalledWith(100)
  })
})
