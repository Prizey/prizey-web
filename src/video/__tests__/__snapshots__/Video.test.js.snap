// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
      "textAlign": "center",
    }
  }
>
  <div
    style={
      Object {
        "height": "100vh",
        "position": "relative",
      }
    }
  >
    <div
      height="100%"
      onEnded={[Function]}
      onProgress={[Function]}
      playing={true}
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
          "zIndex": 1,
        }
      }
      url="Foo videoEmbedUrl"
      width="100%"
    >
      react-player - 
    </div>
    <div
      style={
        Object {
          "color": "white",
          "position": "absolute",
          "textAlign": "center",
          "top": "13%",
          "width": "100%",
          "zIndex": 2,
        }
      }
    >
      Foo videoText
    </div>
    <div
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.1)",
          "bottom": "12%",
          "height": "5px",
          "left": "2.5%",
          "position": "absolute",
          "width": "95%",
          "zIndex": 2,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgb(193, 233, 18)",
            "height": "100%",
            "transitionDuration": "3s",
            "transitionProperty": "width",
            "width": "0%",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "backgroundColor": "tranparent",
          "height": "13%",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 5,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "tranparent",
          "bottom": 0,
          "height": "12%",
          "position": "absolute",
          "width": "100%",
          "zIndex": 5,
        }
      }
    />
  </div>
</div>
`;
